ext/hal/CMakeLists.txt

```diff
@@ -1,5 +1,13 @@
+if(CONFIG_LIBMETAL)
```

Comment:
This libmetal-specific code does not belong in ext/hal/CMakeLists.txt.

It should be pushed into libmetal somehow.

Reply:
How, I don't think it makes sense to modify the CMakeLists.txt file in libmetal.  I don't see a way to do this beyond moving libmetal down another dir, but that seems silly just to avoid a little CMake foo in ext/hal that is clearly related to libmetal.

Reply:
I agree with @SebastianBoe , we should not allow the top level CMakeLists.txt file to be come the place where modules are being configured, this will become unmaintainable if everyone starts putting their logic in there. Going down one level seems reasonable, we did this for a few other modules. Also, the README.libmetal does not belong in there.
All of this seems silly now, but when we move to multi-git structure, it will become necessary.

Reply:
"Going down one level seems reasonable, we did this for a few other modules."
+1

Reply:
Ok, I'll change it, what's an example where we do this one additional level stuff?